app/src/interfaces/list-m2m/list-m2m.vue

```diff
@@ -456,7 +456,19 @@ function editItem(item: DisplayItem) {
 	const junctionPkField = relationInfo.value.junctionPrimaryKeyField.field;
 
 	newItem = false;
-	editsAtStart.value = item;
+
+	relatedPrimaryKey.value = get(item, [junctionField, relatedPkField], null);
+	editsAtStart.value = Object.assign(
+		{},
+		item,
+		relatedPrimaryKey.value
+			? {
+					[junctionField]: {
+						[relatedPkField]: relatedPrimaryKey.value,
+					},
+			  }
+			: {}
+	);
```

Comment:
What's the purpose of thise code? As far as I can see it, this code has no actual effect on `editsAtStart`?

Reply:
As `Object.assign` doesnt merge recursively the nested `[junctionField]: { ... }` is overwriting the full related item object with just the ID

Reply:
Right! Though this still doesn't seem right to me as you then lose all local nested changes when reopening.

Reply:
True re-opening unedited nested items is indeed problematic with this change! am looking into a different approach of clearing the displayData from the editData.

Reply:
Thanks for making sure that I'm not talking ****! 😅🎉

Reply:
The issue with re-opening items turned out to be an unrelated issue also present on the main branch. It took a bit of testing/digging but this seems to be the best solution we got. The function here is to make sure if a related item already exists that only it's primary key is passed to the `drawer-item` because anything contained in this `editsAtStart` object will also be emitted back on save which is causing problems for fields that weren't edited in the process. As for re-opening edits this works perfectly with just the PK as the `drawer-item` itself will fetch all needed values when it receives a relatedPrimaryKey prop but won't emit them back unless they're actually edited.
https://github.com/directus/directus/blob/f77d12eff4b11862415a432ddf3be97a4d4a729e/app/src/views/private/components/drawer-item.vue#L299